### ISSUE TITLE
Feature: Bip39 Wallet

### ISSUE THREAD

**Issue by bonomat** [itchysats/itchysats#3172]
# Implement Bip39 for our wallet

## For existing users 

When importing/creating a new wallet notify the user that the old wallet will be drained.

If there are any open position (also pending close, i.e. the commit tx or CET tx is not yet confirmed), it is not possible to create a new wallet. 

The way how I imagine this is a simple edit field which generated 24 seed words or allows you to edit it to import an existing seed phrase. We did something like this before in our browser extension and this wasn't too bad. 

## For new users

We generate a default wallet (using Bip39) but let the user create a new one through the wallet interface which then goes back to the same functionality as above. That way we don't have to provide a different setup scenario for new users vs existing users.

**Comment by bonomat** [comment]
## Security

Should we generate the seed on the client side or on the daemon side? Since the connection is not encrypted, I guess it does not matter much?

**Comment by holzeis** [comment]
Regarding security. I think we should recommend the use of an encrypted connection. similarly, how we should recommend creating backups 🙂  - but it is mainly an infrastructural topic. and the creation and renewals of certificates should not be in our responsibility. 

That being said I see two ways how an encrypted connection can be supported. Either through an reverse proxy (e.g. nginx or haproxy) or directly on the itchysats daemon. (rocket supports tls - https://rocket.rs/v0.5-rc/guide/configuration/#tls). At home I am running a reverse proxy, which is working fine, but might be too much configuration effort for some. From what I've seen of Start9 this platform is also coming out of the box with an nginx reverse proxy - so I guess we are fine there.

In conclusion, in my opinion we should expect that the client server communication is encrypted. (if its not encrypted - its the users risk).

**Comment by holzeis** [comment]
@bonomat should we be able to display the generated mnemonic on the wallet? I was checking with my Jaxx wallet and it is offering that feature, but it would also mean that we have to store the mnemonic. I don't think that should be a big problem, however it would extend the complexity of this ticket.

**Comment by bonomat** [comment]
> @bonomat should we be able to display the generated mnemonic on the wallet? 

I'm not sure I understand your question: Do you mean: shall we show the Bip39 seed phrase in the frontend ? 
Yes, that's the idea of having the Bip39 wallet because you should be able to back it up.

I think of simple wizard flow like this: 
1. Enter seed (click to create new or enter seed)
2. Confirm seed (enter seed again to ensure you entered it correctly)
3. Create wallet from seed.

**Comment by holzeis** [comment]
@bonomat I am asking because it would mean that we have to store the plain text Bip39 seed phrase to the database. And if we do that I guess we can also skip store the seed file to the disk - as we can generate it on the fly. (by the way, is there a reason why we currently store the seed file to the disk and not to the database?)

**Comment by bonomat** [comment]
Yes, I think storing the seed phrase makes sense. 
I think the initial idea was to have an encrypted seed file and hence went for a file instead of the db.

**Comment by bonomat** [comment]
This project is unmaintained
